File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = s.length();
        if(n != t.length()) return false;

        vector<int> freq(26);

        for(auto& ch:s){
            freq[ch-'a']++;
        }

        for(auto& ch:t){
            freq[ch-'a']--;
        }

        for(auto& num: freq){
            if(num != 0) return false;
        }

        return true;

    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        n = len(s)
        m = len(t)

        if n!=m:
            return False

        freq = [0]*26

        for char in s:
            idx = ord(char) - ord('a')
            freq[idx] += 1
        
        for char in t:
            idx = ord(char) - ord('a')
            if freq[idx] == 0:
                return False
            
            else:
                freq[idx] -= 1

        return True


            